child_compassion/wizard/child_description_de.py

```diff
@@ -128,184 +123,375 @@ def _get_school_info_de(cls, cr, uid, child, case_study, context=None):
             else:
                 string += '.'
         else:
-            string += ' geht nicht in die Schule.'
+            child_age = (
+                date.today() - datetime.strptime(
+                    child.birthdate, DF).date()).days / 365
+            if child_age <= 5:
+                string += ' geht noch nicht in die Schule.'
+            else:
+                string += ' geht nicht in die Schule.'
+
         return string
 
     @classmethod
     def _get_guardians_info_de(cls, cr, uid, child, case_study, context=None):
         ''' Generate the guardian description part. Guardians jobs are
-            also included here.
+            also included here. (comments in child_description_fr)
         '''
+        string = u''
         if not case_study.guardians_ids:
             return ''
         male_values = ['father', 'uncle', 'brother', 'grandfather',
                        'stepfather', 'godfather']
         plur_values = ['friends', 'other relatives', 'foster parents']
-        live_with = []
-        male_guardian = ''
-        female_guardian = ''
+        if child.gender == 'M':
+            prefix = [u'sein', u'seine', u'seine']
+        else:
+            prefix = [u'ihr', u'ihre', u'ihre']
+
+        live_with = OrderedDict()
+        male_guardians = dict()
+        female_guardians = dict()
+        live_in_institut = False
+
         for guardian in case_study.guardians_ids:
-            value = (guardian.value_de if guardian.value_de
-                     else guardian.value_en)
-            if guardian.value_en in male_values:
-                child_to_guardian_pronoun = (u'seinem' if child.gender == 'M'
-                                             else u'ihrem')
-                live_with.append(u'%s %s' % (child_to_guardian_pronoun, value))
-                male_guardian = value if not male_guardian else male_guardian
-            elif guardian.value_en in plur_values:
-                if value == 'friends' or value == 'Freunden':
-                    live_with.append(u'%s %s'
-                                     % (u'seinen' if child.gender == 'M'
-                                        else u'ihren', value))
-                elif (value == 'other relatives'
-                      or value == 'anderen Verwandten'):
-                    live_with.append(u'anderen  %s' % value)
-                elif value == 'foster parents' or value == 'Pflegeeltern':
-                    live_with.append(u'%s %s'
-                                     % (u'seinen' if child.gender == 'M'
-                                        else u'ihren', value))
+            value = guardian.value_de or guardian.value_en
+
+            if guardian.value_en != 'institutional worker':
+                if guardian.value_en in male_values:
+                    male_guardians[guardian.value_en] = value
+                    if guardian.value_en != 'brother':
+                        live_with[guardian.value_en] = u'{} {}'.format(

```

Comment:
Good idea to use string.format! But for more clarity, you shoud include argument index, or key as described in PEP 3101.